src/test/java/com/loofah/graph/api/services/SkillServiceTest.java

```diff
@@ -11,42 +13,60 @@
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
-import static com.loofah.graph.api.helpers.TestHelpers.GRADE_ID;
+import static com.loofah.graph.api.helpers.TestHelpers.*;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @RunWith(MockitoJUnitRunner.class)
 public class SkillServiceTest {
 
     @Mock
-    private DataRetriever skillRetriever;
+    private DataRetriever dataRetriever;
 
     @InjectMocks
     private SkillService skillService;
 
     @Test
-    public void getById() {
+    public void getById_whenSkillExists_thenReturnSkill() {
 
-        Skill expectedSkill = Skill.builder().build();
-        when(skillRetriever.getSkillById("id")).thenReturn(expectedSkill);
+        Skill expectedSkillInDb = getDefaultSkillBuilder().build();
+        SkillDTO expectedSkillDTO = getDefaultSkillDTO();
+        when(dataRetriever.getSkillById("id")).thenReturn(Optional.of(expectedSkillInDb));
+        when(dataRetriever.getCategoryById(expectedSkillInDb.getCategoryId())).thenReturn(Optional.of(expectedSkillDTO.getCategory()));
 
-        Skill actualSkill = skillService.getById("id");
+        SkillDTO actualSkill = skillService.getById("id");
 
-        assertEquals(expectedSkill, actualSkill);
+        assertEquals(actualSkill, expectedSkillDTO);
```

Comment:
Nitpick: shouldn't this be (expected, actual) rather than (actual, expected)?

Reply:
👍 good spot. Have updated